File: job-tracker/src/types/Application.ts
export type AppStatus =
  | 'Not applied'
  | 'Saved'
  | 'Applied'
  | 'Interview'
  | 'Offer'
  | 'Rejected'
  | 'Ghosted';

export interface Application {
  id: number;
  company: string;
  role: string;
  status: AppStatus;
  dateApplied: string;
  salary: string;
  location: string;
  link: string;
  notes: string;
  createdAt: number;
}

File: JobTrackerApi/Models/Application.cs
namespace JobTrackerApi.Models;

public class Application
{
    public int Id { get; set; }
    public string Company { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public string Status { get; set; } = "Applied";
    public string? DateApplied { get; set; }
    public string? Salary { get; set; }
    public string? Location { get; set; }
    public string? Link { get; set; }
    public string? Notes { get; set; }
    public long CreatedAt { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
}